### PR TITLE
Pool chunk slices instead of reusing them

### PR DESCRIPTION
**Pull request by PapaCharlie**
Similar to #2, where buffers are reused instead of pooled. Reusing a buffer instead of pooling it means its capacity is never released. Therefore if a large spike in traffic causes that buffer to grow, the resources it holds onto are never returned even as the traffic decreases. This causes those large buffers to stick around and not get used.

This change uses a `sync.Pool` instead of each `deltaSender` having its own buffer. This way buffers are automatically released when they aren't used.

Here's a screenshot of the motivation behind this change, these unfreed buffers are responsible for about 20% of the memory usage of the observer:
image

### CODE REVIEW

**Review by ZoabKapoor — COMMENTED** [review]
Looks overall reasonable to me! One question -- is that screenshot of 3GB from a local deployment? If yes, can we add a screenshot of how much memory is used **after** this change so we know that it improved things?

**Comment by PapaCharlie** [comment]
No the 3GB screencap is from prod, under loads I just can't reproduce locally, I physically do not have the hardware for it haha. Every time I tried, it's shadowed by the actual ZK data and isn't significant enough to show up.

File: internal/server/handlers_delta.go [hunk 1]
```diff
@@ -87,42 +88,61 @@ type deltaSender struct {
...
+	// empty.
+	queue, ok := queuedUpdatesPool.Get().(*[]queuedResourceUpdate)
+	if ok && cap(*queue) >= size {
+		*queue = (*queue)[:0]
+	} else {
+		if ok {
+			// Return the queue that was too short to the pool
+			queuedUpdatesPool.Put(queue)
+		}
+		queue = new([]queuedResourceUpdate)
```
Comment (bohhyang) [inline internal/server/handlers_delta.go:113]: for my knowledge, will this release the the shorter slice and create a new one in the pool?

**Review by bohhyang — APPROVED** [review]

File: internal/server/handlers_delta.go (hunk 1 above)
Comment (PapaCharlie) [reply internal/server/handlers_delta.go:113]: You can see just above, if the slice we get from the pool is too short, it returns it to the pool. I left a comment on what it means if `ok` is true

**Pull request merged**